neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/query/parsing/impl/Neo4jQueryRendererDelegate.java

```diff
@@ -71,6 +76,35 @@ private void match(StringBuilder queryBuilder, String targetAlias, String label)
 		node( queryBuilder, targetAlias, label );
 	}
 
+	private void optionalMatch(StringBuilder queryBuilder, String targetAlias) {
```

Comment:
What part of the query does this method exactly add?

I'm confused a bit by the general organization of the logic. I don't think that the renderer delegate should "know" the resolver delegate and invoke methods from it. Rather the alias handling should be part of the predicates IMO. The predicate factory could assign the right alias names depending on the "depth" of the predicate to be created. 

"Rendering" here should be more or less the invocation of the predicate tree root node to render itself, which in turn (recursively) asks all children to render themselves (see the call to `builder.build()` in `MongoDBQueryRendererDelegate#getResult`(. So e.g. "rendering" a `Neo4jComparisonPredicate` would yield the complete query part making up that comparison, including the node path in case its an embeddable. 

At least that was the original idea of the mechanism.